### DISCUSSION TITLE
What new or better features needed in e.g. VS2022 to support [NativeAOT] debugging experience?

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [dotnet/runtime#74908] [Ideas] [answered]
Recently, Godot moves from mono embedding using a C++ Main entry to one [NativeAOT] involving C# Main entry with the bulk in c++ Shared Library (`_internal`) 

<img src="https://user-images.githubusercontent.com/49812372/159660795-5694749f-6172-4cee-a415-845fa37a5974.png" width="400" />

[We are improving](https://github.com/godotengine/godot-csharp-visualstudio/issues/36#issuecomment-1233641112) a Godot-specific Visual Studio 2022 **Extension** to improve our experience HOW BEST to debug a NativeAOT with a c# Main entry.
![image](https://user-images.githubusercontent.com/49812372/187818050-735a4ce1-7dc6-44bb-94a2-8f4dda6366f9.png)

Any feedback, suggestions, and curiosity appreciated.

**Comment by wizardars** [comment, marked as answer]
Are you asking about it?

https://github.com/dotnet/runtimelab/blob/feature/NativeAOT/docs/using-nativeaot/debugging.md
